chain/types/event.go

```diff
@@ -32,4 +32,7 @@ type EventEntry struct {
 	Value []byte
 }
 
+// TODO: implement EventEntry#UnmarshalJSON and EventEntry#MarshalJSON to allow for both compact and
+// non-compact forms as per ActorEvent#UnmarshalJSON and ActorEvent#MarshalJSON
+
```

Comment:
We can get rid of this if we make the compact form the only available form right ? The `ActorEvent` encoders and decoders already handle the compaction for Event entries.

Reply:
No, not quite, because there are existing API methods that use this via the `Event` type that we don't use here. Such as `ChainGetEvents`.

What I would do I think is implement an unmarshaller that takes either form.

Also, we should probably also do this same treatment for the filter, currently I'm constructing queries with filters like this:

```json
{"method":"Filecoin.GetActorEvents", "params": [{"fromHeight":1427974,"toHeight":-1,"fields":{"$type":[{"Codec":81,"Value":"cHZlcmlmaWVyLWJhbGFuY2U="},{"Codec":81,"Value":"amFsbG9jYXRpb24="},{"Codec":81,"Value":"cmFsbG9jYXRpb24tcmVtb3ZlZA=="},{"Codec":81,"Value":"ZWNsYWlt"},{"Codec":81,"Value":"bWNsYWltLXVwZGF0ZWQ="},{"Codec":81,"Value":"bWNsYWltLXJlbW92ZWQ="},{"Codec":81,"Value":"bmRlYWwtYWN0aXZhdGVk"},{"Codec":81,"Value":"b2RlYWwtdGVybWluYXRlZA=="},{"Codec":81,"Value":"bmRlYWwtY29tcGxldGVk"},{"Codec":81,"Value":"c3NlY3Rvci1wcmVjb21taXR0ZWQ="},{"Codec":81,"Value":"cHNlY3Rvci1hY3RpdmF0ZWQ="},{"Codec":81,"Value":"bnNlY3Rvci11cGRhdGVk"},{"Codec":81,"Value":"cXNlY3Rvci10ZXJtaW5hdGVk"}]}}],"id":1,"jsonrpc":"2.0"}
```

I'd like to do `[81,"sector-terminated"]`, and it would be nice if that was flexible and took either form too I think

Reply:
FYI, this is an existing API that we probably don't want to break:

```
$ curl -s -X POST   -H "Content-Type: application/json"   --data '{"method":"Filecoin.ChainGetEvents", "params": [{"/":"bafy2bzaceahc33tda6xtwxx7agspupiiwrgjwkniie5x3epz76tstv5nhdn5y"}],"id":1,"jsonrpc":"2.0"}' http://localhost:1235/rpc/v1
{"jsonrpc":"2.0","result":[{"Emitter":6,"Entries":[{"Flags":3,"Key":"$type","Codec":81,"Value":"amFsbG9jYXRpb24="},{"Flags":3,"Key":"id","Codec":81,"Value":"GXxO"},{"Flags":3,"Key":"client","Codec":81,"Value":"GQPz"},{"Flags":3,"Key":"provider","Codec":81,"Value":"GQPo"}]}],"id":1}
```

Reply:
Not super relevant to this PR, but it might be worth just deprecating the uncompacted format -- perhaps renaming the existing methods to `ChainGetEventsLegacy` (so easy breaking change to integrate initially), and then killing them in 3 months. I'm generally opposed to having two different formats for something so user-facing, it's only gonna be more detail for users to wrap their heads around.

Buuutt...future discussion, doesn't affect this changeset.